core/src/main/kotlin/knote/PageManagerImpl.kt

```diff
@@ -66,12 +60,36 @@ internal class PageManagerImpl(
         }
     }
 
-    override fun evalPage(pageId: String): Pair<PageScript, Any?>? {
+    override fun resultType(pageId: String): KType? {
+        val page = findPage(pageId) ?: run {
+            logger.warn("could not find page $pageId")
+            return null
+        }
+        val pageScript = page.compiledScript ?: run {
+            logger.warn("script for page $pageId is not compiled")
+            evalPage(pageId)?.compiledScript ?: return null
+        }
+        val processFunction =
+            pageScript::class.declaredMemberFunctions.find { it.name == "process" } ?: run {
+                logger.error("no function `process` found in $pageId")
+                return null
+            }
+        return processFunction.returnType
+    }
```

Comment:
oh thank god we needed this for sure. I'll have to update this on the GUI accordingly when the merge goes up